systematicstools/interface/ISystProviderTool.cc

```diff
@@ -110,6 +110,28 @@ bool ISystProviderTool::ConfigureFromParameterHeaders(
 }
 
 #ifndef NO_ART
+
+systtools::event_unit_response_t
+ISystProviderTool::GetDefaultEventResponse() {
+  systtools::SystMetaData const &smd = this->GetSystMetaData();
+
+  systtools::event_unit_response_t resp;
+  for( auto &sph : smd ){
+
+    resp.push_back({sph.systParamId, {}});
+    if (sph.isCorrection) {
+      resp.back().responses.push_back(1.);
+    } else {
+      for ( std::vector<double>::size_type i_var=0; i_var<sph.paramVariations.size(); i_var++) {
+        resp.back().responses.push_back(1.);
+      }
+    }
+
+  }
+  return resp;
+}
```

Comment:
This member function is more complicated than it needs to be.  I suggest a simpler implementation, broken into two functions:

```c++
ParamResponses 
responses_for(SystParamHeader const& sph)
{
  if (sph.isCorrection) {
    return {sph.systParamId, std::vector{1.}};
  }
  return {sph.systParamId, std::vector<double>(sph.paramVariations.size(), 1.)};
}

systtools::event_unit_response_t
ISystProviderTool::GetDefaultEventResponse() const 
{
  auto const& smd = this->GetSystMetaData();
  systtools::event_unit_response_t resp;
  resp.reserve(smd.size());
  for(auto const& sph : smd) {
    resp.push_back(response_for(sph));
  }
  return resp;
}
```

Reply:
Thank you! This is added in the new commit.